src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/SealInternalTypesTests.cs

```diff
@@ -0,0 +1,376 @@
+        public Task GenericClass_WithSubclass_NoDiagnostic_CS(string baseClass, string derivedClass)
+        {
+            string source = $@"
+internal class {baseClass}
+internal sealed class {derivedClass}";
+
+            return VerifyCS.VerifyCodeFixAsync(source, source);
+        }
+
+        [Theory]
+        [InlineData("B(Of T)", "D : Inherits B(Of Integer)")]
+        [InlineData("B(Of T)", "D(Of T) : Inherits B(Of T)")]
+        [InlineData("B(Of T, U)", "D(Of T) : Inherits B(Of T, Integer)")]
+        public Task GenericClass_WithSubclass_NoDiagnostic_VB(string baseClass, string derivedClass)
+        {
+            string source = $@"
+Friend Class {baseClass}
+End Class
+
+Friend NotInheritable Class {derivedClass}
+End Class";
+
+            return VerifyVB.VerifyCodeFixAsync(source, source);
+        }
+
+        [Fact]
+        public Task PartialClass_ReportedAndFixedAtAllLocations_CS()
+        {
+            var test = new VerifyCS.Test
+            {
+                TestState =
+                {
+                    Sources =
+                    {
+                        @"internal class Base { }",
+                        @"internal partial class {|#0:Derived|} : Base { }",
+                        @"internal partial class {|#1:Derived|} : Base { }"
+                    },
+                    ExpectedDiagnostics =
+                    {
+                        VerifyCS.Diagnostic(Rule).WithArguments("Derived").WithLocation(0).WithLocation(1)
+                    }
+                },
+                FixedState =
+                {
+                    Sources =
+                    {
+                        @"internal class Base { }",
+                        @"internal sealed partial class Derived : Base { }",
+                        @"internal sealed partial class Derived : Base { }"
+                    }
+                }
+            };
+            return test.RunAsync();
+        }
+
+        [Fact]
+        public Task PartialClass_ReportedAndFixedAtAllLocations_VB()
+        {
+            var test = new VerifyVB.Test
+            {
+                TestState =
+                {
+                    Sources =
+                    {
+                        @"
+Friend Class Base
+End Class", @"
+Partial Friend Class {|#0:Derived|} : Inherits Base
+End Class", @"
+Partial Friend Class {|#1:Derived|} : Inherits Base
+End Class"
+                    },
+                    ExpectedDiagnostics =
+                    {
+                        VerifyVB.Diagnostic(Rule).WithArguments("Derived").WithLocation(0).WithLocation(1)
+                    }
+                },
+                FixedState =
+                {
+                    Sources =
+                    {
+                        @"
+Friend Class Base
+End Class", @"
+Partial Friend NotInheritable Class Derived : Inherits Base
+End Class", @"
+Partial Friend NotInheritable Class Derived : Inherits Base
+End Class"
```

Comment:
I'm concerned about adding attempting to seal a partial in a generated file. I'm not sure how other analyzers deal with similar scenarios. @Evangelink Have you got any idea?

cc @sharwell as well.

Reply:
@NewellClark I think if one part of the symbol is in non-generated code, we'll get the symbol reported. Testing this should be easy, just add multiple sources, with one of them having `// <autogenerated>` at the top.

Reply:
Sorry for the late answer! I don't know but I think adding the test should be easy and will document the behavior anyways.